vm/src/frame.rs

```diff
@@ -659,9 +659,30 @@ impl ExecutingFrame<'_> {
             }
             bytecode::Instruction::UnpackSequence { size } => {
                 let value = self.pop_value();
-                let elements = vm.extract_elements(&value)?;
-                if elements.len() != *size {
-                    Err(vm.new_value_error("Wrong number of values to unpack".to_owned()))
+                let elements = vm.extract_elements(&value).map_err(|e| {
+                    if e.class().is(&vm.ctx.exceptions.type_error) {
+                        vm.new_type_error(format!(
+                            "cannot unpack non-iterable {} object",
+                            value.class().name
+                        ))
+                    } else {
+                        e
+                    }
+                })?;
+                let ordering = elements.len().cmp(size);
+                if ordering != std::cmp::Ordering::Equal {
```

Comment:
Maybe you could have the `for element in ...` be a match arm of `match ordering` to avoid the unsafe unreachable hint?

Reply:
Thanks, it looks better